### PR TITLE
added phpdocs to InternalRequest

### PR DESCRIPTION
**Pull request by staabm**
to support IDE autocompletion

### CODE REVIEW

File: lib/InternalRequest.php [hunk 1]
```diff
@@ -7,28 +7,51 @@
...
     public $responseWriter;
+    /** @var array */
     public $badFilterKeys = [];
+    /** @var boolean */
     public $filterErrorFlag;
+    /** @var boolean */
     public $isServerStopping;
+    /** @var string */
     public $streamId;
     public $trace;
```
Comment (staabm) [inline lib/InternalRequest.php]: I couldnt figure out what the correct type should be for `$trace`

Comment (bwoebi) [inline lib/InternalRequest.php]: string.

Comment (staabm) [inline lib/InternalRequest.php]: fixed

File: lib/InternalRequest.php [hunk 2]
```diff
@@ -7,28 +7,52 @@
 class InternalRequest {
     use Struct;
 
+    /** @var Client */
     public $client;
+    /** @var Vhost */
     public $vhost;
+    /** @var callable */
```
Comment (staabm) [inline lib/InternalRequest.php]: is `callable` fine?

Comment (bwoebi) [inline lib/InternalRequest.php]: yes, callable or null

**Comment by staabm** [comment]
here we go

File: lib/InternalRequest.php [hunk 3]
```diff
@@ -7,28 +7,52 @@
...
+    /** @var \Generator */
     public $responseWriter;
+    /** @var array */
     public $badFilterKeys = [];
+    /** @var boolean */
     public $filterErrorFlag;
+    /** @var boolean */
     public $isServerStopping;
+    /** @var string */
     public $streamId;
```
Comment (bwoebi) [inline lib/InternalRequest.php]: streamId is integer

File: lib/InternalRequest.php [hunk 4]
```diff
@@ -7,28 +7,52 @@
...
+    /** @var Body */
     public $body;
+    /** @var string */
     public $uri;
+    /** @var string */
     public $uriRaw;
+    /** @var string */
     public $uriHost;
+    /** @var string */
     public $uriPort;
```
Comment (bwoebi) [inline lib/InternalRequest.php]: integer

File: lib/InternalRequest.php (hunk 3 above)
Comment (staabm) [inline lib/InternalRequest.php]: I somehow had in mind that we discussed that this needs to be a string so it can "never" overflow?

Comment (bwoebi) [inline lib/InternalRequest.php]: no, that's not stream ids, that's about ping counter.

Comment (kelunik) [inline lib/InternalRequest.php]: They're per connection? When do I use these IDs?

Comment (bwoebi) [inline lib/InternalRequest.php]: you don't, it's an internal detail to HTTP/2

Comment (staabm) [inline lib/InternalRequest.php]: changed

File: lib/InternalRequest.php [hunk 5]
```diff
@@ -7,28 +7,52 @@
 class InternalRequest {
     use Struct;
 
+    /** @var Client */
     public $client;
```
Comment (staabm) [inline lib/InternalRequest.php]: @bwoebi should those fields be properly typed instead of using phpdoc?

Comment (bwoebi) [inline lib/InternalRequest.php]: what do you mean with properly typed here?

Comment (staabm) [inline lib/InternalRequest.php]: use `public Client $client` instead of a phpdoc (where appropriate in this class)

Comment (kelunik) [inline lib/InternalRequest.php]: We don't have typed properties yet.

Comment (staabm) [inline lib/InternalRequest.php]: its a pity that I spent so less time with php7 that I even dont know what the language actually supports. :-/

**Pull request merged**